src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACTerminalUnitVariableRefrigerantFlow.cpp

```diff
@@ -432,6 +434,18 @@ namespace energyplus {
       }
     }
 
+    // Design Specification Multispeed Object Name
```

Comment:
I don't see any FT testing for this one?

Reply:
@jmarrec I didn't get too far on this PR yet. I sort of got tripped up on FT. Appears that for AirLoopHVACUnitarySystem, it FTs the UnitarySystemPerformanceMultispeed object, but also _creates_ one if one doesn't exist. Under what conditions should we create one for ZoneHVACWaterToAirHeatPump and ZoneHVACTerminalUnitTVariableRefrigerantFlow?

Reply:
AirLoopHVACUnitarySystem I believe throws if you use multispeed / variable speed coils without a UnitarySystemPerformanceMultispeed object, hence why the FT creates one if needed.

I'm not 100% sure whether the VRF TU and the ZoneHVAC:WaterToAirHeatPump have the same behavior now. If so, we should create one in FT. In that case, should move the default creation to a helper function so we can reuse in other modules.

Reply:
oh, that's quite confusing, but it seems that this applies not to the coils but to the fans. The original NFP proposed a new object "FanPerformance:Multispeed", but apparently UnistarySystemPerformance:Multispeed was kept.

https://github.com/NREL/EnergyPlus/blob/develop/design/FY2023/NFP_MultispeedFans.md

My understanding is that this is purely optional, regardless of the coils you use for the ZoneHVAC:WAHP or ZoneHVAC:TU:VRF objects. We should probably make some dummy IDF tests to confirm. I'll ask the E+ dev team right now too.

Reply:
@jmarrec I stubbed UnitarySystemPerformance FT to handle additional coil types `CoilXXXWaterToAirHeatPumpEquationFit`, `CoilXXXDXVariableRefrigerantFlow`, and `CoilXXXDXVariableRefrigerantFlowFluidTemperatureControl`. But I'm a little unclear on how to determine `NumberofSpeedsforXXX`. The speeds should not correspond to the coils, but to the supply fan? Are we going to need to create a new setter method for speeds on the UnitarySystemPerformance object?

Reply:
My earlier post was misleading I think. The object varies the fan airflow in conjunction with the heating/cooling capacity.
So Number of Speeds for Heating should match the heating coil number of speeds, when the coil does have speeds.

I don't think we need to create a UnitarySystemPerformance object for VRF / ZoneHVACWaterToAirHeatPump if they don't throw when one isn't present. In that case, we're just left translating whatever the user **chose** to specify.


Also, take `VRFMultispeedFan.idf`for eg. You have one `ZoneHVAC:TerminalUnit:VariableRefrigerantFlow` (`TU1`) object with some `COIL:Cooling:DX:VariableRefrigerantFlow` and `COIL:Heating:DX:VariableRefrigerantFlow`. These coils do not have speed-related fields. Yet there is a UnitarySystemPerformance assigned to it.

Reply:
I'm looking at `VRFMultispeedFan.idf`...
```
  UnitarySystemPerformance:Multispeed,
    Sys 1 WaterToAirHP System MultiSpeed Performance,  !- Name
    2,                       !- Number of Speeds for Heating
    2,                       !- Number of Speeds for Cooling
```
Where does the "2" come from? Should this value be hardcoded when coil object types are `COIL:*:DX:VariableRefrigerantFlow`?

Reply:
That is arbitrary I believe.